### PR TITLE
feat: taosadapter support schemaless with ttl

### PR DESCRIPTION
**Pull request by sunpe**

### CODE REVIEW

File: plugin/statsd/config.go [hunk 1]
```diff
@@ -104,4 +106,8 @@ func init() {
 	_ = viper.BindEnv("statsd.deleteTimings", "TAOS_ADAPTER_STATSD_DELETE_TIMINGS")
 	pflag.Bool("statsd.deleteTimings", true, `statsd delete timing cache after gather. Env "TAOS_ADAPTER_STATSD_DELETE_TIMINGS"`)
 	viper.SetDefault("statsd.deleteTimings", true)
+
+	_ = viper.BindEnv("statsd.ttl", "TAOS_ADAPTER_STATSD_TTL")
+	pflag.Bool("statsd.ttl", true, `statsd data ttl. Env "TAOS_ADAPTER_STATSD_TTL"`)
```
Comment (huskar-t) [inline plugin/statsd/config.go:111]: Bool -> Int

File: plugin/opentsdbtelnet/config.go [hunk 2]
```diff
@@ -70,4 +72,8 @@ func init() {
 	_ = viper.BindEnv("opentsdb_telnet.flushInterval", "TAOS_ADAPTER_OPENTSDB_TELNET_FLUSH_INTERVAL")
 	pflag.Duration("opentsdb_telnet.flushInterval", time.Duration(0), `opentsdb_telnet flush interval (0s means not valid) . Env "TAOS_ADAPTER_OPENTSDB_TELNET_FLUSH_INTERVAL"`)
 	viper.SetDefault("opentsdb_telnet.flushInterval", time.Duration(0))
+
+	_ = viper.BindEnv("opentsdb_telnet.ttl", "TAOS_ADAPTER_OPENTSDB_TELNET_TTL")
+	pflag.Duration("opentsdb_telnet.ttl", 0, `opentsdb_telnet data ttl. Env "TAOS_ADAPTER_OPENTSDB_TELNET_TTL"`)
```
Comment (huskar-t) [inline plugin/opentsdbtelnet/config.go:77]: Duration->Int

File: plugin/opentsdb/plugin.go [hunk 3]
```diff
@@ -146,6 +152,13 @@ func (p *Plugin) insertTelnet(c *gin.Context) {
 		p.errorResponse(c, http.StatusBadRequest, errors.New("db required"))
 		return
 	}
+
+	var ttl int
+	ttlStr := c.Param("ttl")
+	if len(ttlStr) > 0 {
+		ttl, _ = strconv.Atoi(ttlStr)
```
Comment (huskar-t) [inline plugin/opentsdb/plugin.go:159]: handle error

File: plugin/opentsdb/plugin.go [hunk 4]
```diff
@@ -94,6 +95,11 @@ func (p *Plugin) insertJson(c *gin.Context) {
 		p.errorResponse(c, http.StatusBadRequest, err)
 		return
 	}
+	var ttl int
+	ttlStr := c.Param("ttl")
+	if len(ttlStr) > 0 {
+		ttl, _ = strconv.Atoi(ttlStr)
```
Comment (huskar-t) [inline plugin/opentsdb/plugin.go:101]: handle error

File: plugin/opentsdb/plugin.go [hunk 5]
```diff
@@ -146,6 +152,13 @@ func (p *Plugin) insertTelnet(c *gin.Context) {
 		p.errorResponse(c, http.StatusBadRequest, errors.New("db required"))
 		return
 	}
+
+	var ttl int
+	ttlStr := c.Param("ttl")
```
Comment (huskar-t) [inline plugin/opentsdb/plugin.go:157]: 用 param 还是 query?

File: plugin/opentsdb/plugin.go [hunk 6]
```diff
@@ -94,6 +95,11 @@ func (p *Plugin) insertJson(c *gin.Context) {
 		p.errorResponse(c, http.StatusBadRequest, err)
 		return
 	}
+	var ttl int
+	ttlStr := c.Param("ttl")
```
Comment (huskar-t) [inline plugin/opentsdb/plugin.go:99]: 用 param 还是 query?

File: plugin/nodeexporter/config.go [hunk 7]
```diff
@@ -99,4 +100,8 @@ func init() {
 	_ = viper.BindEnv("node_exporter.gatherDuration", "TAOS_ADAPTER_NODE_EXPORTER_GATHER_DURATION")
 	pflag.Duration("node_exporter.gatherDuration", 5*time.Second, `node_exporter gather duration. Env "TAOS_ADAPTER_NODE_EXPORTER_GATHER_DURATION"`)
 	viper.SetDefault("node_exporter.gatherDuration", "5s")
+
+	_ = viper.BindEnv("node_exporter.ttl", "TAOS_ADAPTER_NODE_EXPORTER_TTL")
+	pflag.Duration("node_exporter.ttl", 0, `node_exporter data ttl. Env "TAOS_ADAPTER_NODE_EXPORTER_TTL"`)
```
Comment (huskar-t) [inline plugin/nodeexporter/config.go:105]: Duration -> Int

File: plugin/opentsdb/plugin.go [hunk 8]
```diff
@@ -94,6 +95,11 @@ func (p *Plugin) insertJson(c *gin.Context) {
 		p.errorResponse(c, http.StatusBadRequest, err)
 		return
 	}
+	var ttl int
```
Comment (huskar-t) [inline plugin/opentsdb/plugin.go:98]: ttl 的测试呢

File: plugin/opentsdbtelnet/plugin.go [hunk 9]
```diff
@@ -299,7 +299,7 @@ func (p *Plugin) handleData(index int, line []string) {
 		start = time.Now()
 	}
 	logger.Debugln(start, " insert telnet payload ", line)
-	err = inserter.InsertOpentsdbTelnetBatch(taosConn.TaosConnection, line, p.conf.DBList[index])
+	err = inserter.InsertOpentsdbTelnetBatch(taosConn.TaosConnection, line, p.conf.DBList[index], p.conf.TTL)
```
Comment (huskar-t) [inline plugin/opentsdbtelnet/plugin.go:302]: 需要增加测试

File: plugin/statsd/plugin.go [hunk 10]
```diff
@@ -132,7 +132,7 @@ func (p *Plugin) HandleMetrics(serializer *influx.Serializer, metric telegraf.Me
 
 	start := time.Now()
 	logger.Debugln(start, "insert line", string(data))
-	result, err := inserter.InsertInfluxdb(taosConn.TaosConnection, data, p.conf.DB, "ns")
+	result, err := inserter.InsertInfluxdb(taosConn.TaosConnection, data, p.conf.DB, "ns", p.conf.TTL)
```
Comment (huskar-t) [inline plugin/statsd/plugin.go:135]: 需要增加测试

File: plugin/statsd/config.go (hunk 1 above)
Comment (sunpe) [reply plugin/statsd/config.go]: done

File: plugin/opentsdbtelnet/config.go (hunk 2 above)
Comment (sunpe) [reply plugin/opentsdbtelnet/config.go]: done

File: plugin/opentsdb/plugin.go (hunk 4 above)
Comment (sunpe) [reply plugin/opentsdb/plugin.go]: done

File: plugin/opentsdb/plugin.go (hunk 6 above)
Comment (sunpe) [reply plugin/opentsdb/plugin.go]: done

File: plugin/nodeexporter/config.go (hunk 7 above)
Comment (sunpe) [reply plugin/nodeexporter/config.go]: done

File: plugin/collectd/plugin_test.go [hunk 11]
```diff
@@ -69,27 +78,20 @@ func TestCollectd(t *testing.T) {
...
-			t.Error(errors.NewError(code, errStr))
-		}
-		wrapper.TaosFreeResult(r)
+		//r := wrapper.TaosQuery(conn, "drop database if exists collectd")
+		//code := wrapper.TaosError(r)
+		//if code != 0 {
+		//	errStr := wrapper.TaosErrorStr(r)
+		//	t.Error(errors.NewError(code, errStr))
+		//}
+		//wrapper.TaosFreeResult(r)
```
[the remark this answers is not in the document]
Comment (sunpe) [reply plugin/collectd/plugin_test.go]: done

**Review by huskar-t — APPROVED** [review]
LGTM

**Pull request merged**